sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -1179,3 +1180,128 @@ private[client] class Shim_v2_1 extends Shim_v2_0 {
 private[client] class Shim_v2_2 extends Shim_v2_1
 
 private[client] class Shim_v2_3 extends Shim_v2_1
+
+private[client] class Shim_v3_1 extends Shim_v2_3 {
+  // Spark supports only non-ACID operations
+  protected lazy val isAcidIUDoperation = JBoolean.FALSE
```

Comment:
@dongjoon-hyun, isn't it `isAcid` defined at `Shim_v0_14`? Was wondering why this was separate variable again. Do you see any possibility that this is different specifically for 3.1? Then, it's fine.

Reply:
Historically, `isAcidIUDoperation` is an evolved one from `isAcid`.

In Hive code, `isAcid` was a general ACID operation and `isAcidIUDoperation` is now used for ACID Insert/Update/Delete operations. Also, they checks `isFullAcidTable` and use it together like this.
```java
else if(!isAcidIUDoperation && isFullAcidTable) {
    destPath = fixFullAcidPathForLoadData(loadFileType, destPath, txnId, stmtId, tbl);
}
```

We don't know the future of Hive. So, for the different parameter name, we had better handle differently. That was my logic.